src/hotspot/cpu/x86/x86.ad

```diff
@@ -3879,6 +3880,18 @@ instruct roundD_reg(legRegD dst, legRegD src, immU8 rmode) %{
   ins_pipe(pipe_slow);
 %}
 
+instruct roundD_reg_avx(legRegD dst, legRegD src, immU8 rmode) %{
+  predicate(UseAVX > 0);
+  match(Set dst (RoundDoubleMode src rmode));
+  format %{ "roundsd $dst,$src" %}
+  ins_cost(150);
+  ins_encode %{
+    __ vroundsd($dst$$XMMRegister, $src$$XMMRegister, $src$$XMMRegister, $rmode$$constant);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+/*
 instruct roundD_mem(legRegD dst, memory src, immU8 rmode) %{
```

Comment:
Don't we want roundD_mem enabled, for both roundsd (UseAVX == 0) and vroundsd (UseAVX > 0)?

Reply:
@dean-long the roundD_mem instruct is the cause of slow performance due to a false dependency. It generates the instruction of the following form which has a 128 bit result:

roundsd xmm0, memory_src, mode
vroundsd xmm0, xmm0, memory_src, mode

xmm0 bits 0:63 are result of round operation on memory_src
xmm0 bits 64:128 are dependent on old value of xmm0  (false dependency)

By forcing the load of memory_src into a register before the operation by as below removes the false dependency:
vmovsd xmm0, memory_src   ; bits 64 and above are cleared by vmovsd
vroundsd xmm0, xmm0, xmm0, mode